pkg/bundle/publish/publish.go

```diff
@@ -185,223 +174,187 @@ func (o *Options) Run(ctx context.Context, cmd *cobra.Command, f kcmdutil.Factor
 
 	var (
 		errs []error
-		// Mappings for mirroring image types.
-		genericMappings []imgmirror.Mapping
-		releaseMappings []imgmirror.Mapping
-		catalogMappings []imgmirror.Mapping
-
 		// Map of remote layer digest to the set of paths they should be fetched to.
-		missingLayers = map[string][]string{}
-		aerr          = &ErrArchiveFileNotFound{}
+		aerr  = &ErrArchiveFileNotFound{}
+		icsps [][]byte
 	)
 
-	for imageName, assoc := range assocs {
-
-		assoc := assoc
-		logrus.Debugf("reading assoc: %s", assoc.Name)
+	// Create target dir for manifest directory
+	manifestDir, err := o.createManifestDir()
+	if err != nil {
+		return err
+	}
 
-		// All manifest layers will be pulled below if associated,
-		// so just sanity-check that the layers are referenced in some association.
-		if len(assoc.ManifestDigests) != 0 {
-			for _, manifestDigest := range assoc.ManifestDigests {
-				if _, hasManifest := assocs[manifestDigest]; !hasManifest {
-					errs = append(errs, fmt.Errorf("image %q: expected associations to have manifest %s but was not found", imageName, manifestDigest))
+	for _, imageName := range assocs.Keys() {
+
+		genericMappings := []imgmirror.Mapping{}
+		releaseMapping := imgmirror.Mapping{}
+		missingLayers := map[string][]string{}
+		icspMapping := make(map[imagesource.TypedImageReference]imagesource.TypedImageReference)
+
+		values, _ := assocs.Search(imageName)
+
+		// Create temp workspace for image processing
+		cleanup, err := o.mktempDir()
```

Comment:
I would just `if !o.SkipCleanup { defer cleanup() }` here since a cond branch that runs cleanup might never be entered.

Reply:
Also this overwrites `o.tmp`, which is an unexpected mutation that could lead to bugs later. I'd rewrite this function to return the dir name, or use the preexisting `o.tmp`.